simtools/model/array_model.py

```diff
@@ -1,19 +1,473 @@
+        self._configFileDirectory = io.getModelOutputDirectory(self._filesLocation, self.label)
+        if not self._configFileDirectory.exists():
+            self._configFileDirectory.mkdir(parents=True, exist_ok=True)
+            self._logger.info('Creating directory {}'.format(self._configFileDirectory))
+        return
+
+    def _buildArrayModel(self):
+        '''
+        Build the site parameters and the list of telescope models,
+        including reading the parameters from the DB.
+        '''
+
+        # Getting site parameters from DB
+        db = db_handler.DatabaseHandler(self._logger.name)
+        self._siteParameters = db.getSiteParameters(
+            self.site,
+            self.modelVersion,
+            self._configFileDirectory,
+            onlyApplicable=True
+        )
+
+        # Building telescope models
+        self._telescopeModel = list()  # List of telescope models
+        _allTelescopeModelNames = list()  # List of telescope names without repetition
+        _allParsToChange = dict()
+        for tel in self.layout:
+            telSize = tel.getTelescopeSize()
+
+            # Collecting telescope name and pars to change from arrayConfigData
+            telModelName, parsToChange = self._getSingleTelescopeInfoFromArrayConfig(
+                tel.name,
+                telSize
+            )
+            if len(parsToChange) > 0:
+                _allParsToChange[tel.name] = parsToChange
+
+            self._logger.debug('TelModelName: {}'.format(telModelName))
+
+            # Building the basic models - no pars to change yet
+            if telModelName not in _allTelescopeModelNames:
+                # First time a telescope name is built
+                _allTelescopeModelNames.append(telModelName)
+                telModel = TelescopeModel(
+                    telescopeName=telModelName,
+                    version=self.modelVersion,
+                    label=self.label,
+                    modelFilesLocations=self._modelFilesLocations,
+                    filesLocation=self._filesLocation
+                )
+            else:
+                # Telescope name already exists.
+                # Finding the TelescopeModel and copying it.
+                for tel in self._telescopeModel:
+                    if tel.telescopeName != telModelName:
+                        continue
+                    self._logger.debug(
+                        'Copying tel model {} already loaded from DB'.format(tel.telescopeName)
+                    )
+                    telModel = copy(tel)
+                    break
+
+            self._telescopeModel.append(telModel)
+
+        # Checking whether the size of the telescope list and the layout match
+        if len(self._telescopeModel) != len(self.layout):
+            self._logger.warning(
+                'Size of telModel does not match size of layout - something is wrong!'
```

Comment:
Still trying to understand what a telModel and a layout is in this language (maybe from this comment you can see that the error message is too cryptic?).

The comment to the class ArrayModel (line 22) says: "It contains the list of TelescopeModel's and an LayoutArray". So with size of telModel, you mean the length of the list of telescope models? Or actually the lenght of the telescope model? I would understand the first, but not the second point.

Reply:
I mean the length of the list of telescope models does not match the number of telescopes in the LayoutArray object.
I expanded the comment to make it clearer.

Reply:
OK!